### PR TITLE
Include disabled secrets in the master-key rotation inventory

### PR DESCRIPTION
**Pull request by CybotTM**
Closes #286.

## What

`vault:rotate-master-key` built its inventory with `findIdentifiers()` (no filters) and loaded each secret with `findByIdentifier()` — both hidden-restricted, so disabled secrets never reached the re-wrap loop. Their DEKs stayed wrapped under the old master key, the very key the command's next-steps output tells the operator to "securely archive or destroy". A secret re-enabled after that is permanently undecryptable. With only disabled secrets in the vault, the command reported "No secrets found" and exited successfully.

## How

All three lookups now use the disabled-visible repository paths the repository already provides for exactly this class of operation (its docblock names "re-enabling it, rotating it, deleting it"):

- inventory: `findIdentifiers(new SecretFilters(includeDisabled: true))`
- pre-flight smoke test and `rotateOne()`: `findByIdentifierIncludingDisabled()`

Safety of the write path was verified before changing the reads: `SecretRepository::save()` persists via a raw `Connection::update(... WHERE uid)` (no restrictions) and the `Secret` model round-trips the `hidden` flag through `toDatabaseRow()`, so rotating a disabled secret re-wraps its DEK without re-enabling it.

## Tests

- Functional (`MasterKeyRotationTest::rotationReWrapsDisabledSecretsSoTheyDecryptAfterReEnabling`): stores an active and a disabled secret, runs the real `vault:rotate-master-key` command, asserts the inventory says "Found 2 secret(s)", switches the provider key, then reads the active secret and re-enables + reads the disabled one — both plaintexts intact. Fails on the unfixed command with "Found 1 secret(s)" (verified A/B via stash).
- Unit: the command test doubles now stub `findByIdentifierIncludingDisabled()`, and `successfulRotationWithConfirm` pins the disabled-visible inventory contract (`includeDisabled === true`).

## Checked, deliberately not changed

`VaultCleanupOrphansCommand` scans via `vaultService->list()` (default `includeDisabled: false`), so disabled orphans are never examined — but that command **deletes** what it classifies as orphaned. Including disabled secrets there would make a deliberately-parked secret auto-deletable, a separate behavior decision with its own risk trade-off; the current skip is fail-safe. Noted in #286.

### CODE REVIEW

File: Tests/Functional/Crypto/MasterKeyRotationTest.php [hunk 1]
```diff
@@ -523,6 +523,57 @@ public function secondRotationKeepsSecretsDecryptableAndAuditChainValid(): void
...
+        $commandTester = new CommandTester($this->get(VaultRotateMasterKeyCommand::class));
+
+        $activeIdentifier = $this->generateUuidV7();
+        $disabledIdentifier = $this->generateUuidV7();
+        $vaultService->store($activeIdentifier, 'active-value');
+        $vaultService->store($disabledIdentifier, 'disabled-value');
+        $vaultService->setEnabled($disabledIdentifier, false, 'out of service during rotation');
+
+        $newKeyPath = $this->instancePath . self::NEW_KEY_FILENAME;
+        file_put_contents($newKeyPath, sodium_crypto_secretbox_keygen());
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply Tests/Functional/Crypto/MasterKeyRotationTest.php:546]: Verified against the code under test: `VaultRotateMasterKeyCommand::loadKeyFromFile()` handles raw binary keys explicitly — its resolution order is documented for exactly this corner case ("never trim() a value that is already binary key material, or a valid 32-byte raw key whose first/last byte happens to be a trim char … is silently corrupted") and step 3 returns the untrimmed raw 32-byte contents, so a trim-char boundary byte cannot flake. `handlesKeyWithTrailingNewline` and `handlesBase64EncodedKeys` pin that behaviour. Raw keygen bytes are also the established pattern of every other test in this file (lines 118, 252, 476, 496). Declining.

File: Tests/Unit/Command/VaultRotateMasterKeyCommandTest.php [hunk 2]
```diff
@@ -336,12 +337,18 @@ public function successfulRotationWithConfirm(): void
...
         $secret = $this->createTestSecret('test-secret');
 
+        // The inventory must be built disabled-visible (#286): a disabled
+        // secret's DEK is wrapped under the key being retired too.
         $this->secretRepository
+            ->expects(self::once())
             ->method('findIdentifiers')
+            ->with(self::callback(
+                static fn (SecretFilters $filters): bool => $filters->includeDisabled,
+            ))
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply Tests/Unit/Command/VaultRotateMasterKeyCommandTest.php:347]: Verified against `SecretFilters`: `$includeDisabled` is a promoted readonly `public bool` and the callback parameter is typed `SecretFilters` with a `: bool` return, so a non-boolean truthy value cannot reach or leave the callback — `=== true` would be a no-op the type system already guarantees. Declining.

File: Tests/Functional/Crypto/MasterKeyRotationTest.php [hunk 3]
```diff
@@ -523,6 +523,57 @@
...
+            'a secret disabled during the rotation must decrypt under the new key once re-enabled',
+        );
+
+        // Cleanup
+        foreach ([$activeIdentifier, $disabledIdentifier] as $identifier) {
+            $vaultService->delete($identifier, self::REASON_TEST_CLEANUP);
+        }
+        if (file_exists($newKeyPath)) {
+            // nosemgrep: php.lang.security.unlink-use.unlink-use - test-owned path
+            unlink($newKeyPath);
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply Tests/Functional/Crypto/MasterKeyRotationTest.php:573]: Intentional test input: `$newKeyPath` is built from `$this->instancePath` + a class constant and guarded by `file_exists` — no user input reaches `unlink()`. Dismissed alert 283 as "used in tests", matching the precedent of alert 228 on the same pattern in this file (line 521).

[the remark this answers is not in the document]
Comment (CybotTM) [reply Tests/Functional/Crypto/MasterKeyRotationTest.php:573]: Same finding as the already-dismissed alert 283 — the rebase shifted the line, minting a new alert. `$newKeyPath` is `$this->instancePath` + a class constant, guarded by `file_exists`; no user input reaches `unlink()`. Dismissed alert 286 as "used in tests" (precedent: alerts 228, 283).
